tests/test_new_return_types_qnode.py

```diff
@@ -0,0 +1,1665 @@
+# Copyright 2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Unit tests for the new return types with QNode.
+"""
+import pytest
+
+import numpy as np
+import pennylane as qml
+
+wires = [2, 3, 4]
+devices = ["default.qubit", "default.mixed"]
+
+
+class TestIntegrationSingleReturn:
+    """Test that single measurements return behavior does not change."""
+
+    @pytest.mark.parametrize("wires", wires)
+    def test_state_default(self, wires):
+        """Return state with default.qubit."""
+        qml.enable_return()
+
+        dev = qml.device("default.qubit", wires=wires)
+
+        def circuit(x):
+            qml.Hadamard(wires=[0])
+            qml.CRX(x, wires=[0, 1])
+            return qml.state()
+
+        qnode = qml.QNode(circuit, dev)
+        res = qnode(0.5)
+
+        qml.disable_return()
+
```

Comment:
For all of these tests, there might be an issue with calling `qml.enable_return()` at the start of the test and calling `qml.disable_return()` at the end.

I'm thinking of the scenario where the test fails before the `qml.disable_return()` is called. Then, if I'm not mistaken, pytest will continue running the other tests (including other test files) with the new return types enabled since the proper clean-up is not run in this scenario.

Reply:
Sounds like a good use case for text fixtures :)

Reply:
I created a folder with all the new return type tests and added a conftest.py file which runs `enable_return` at the start of the session and `disable_return` at the end.